src/lib/utilities/Dialog/Dialog.svelte

```diff
@@ -1,156 +1,177 @@
-	export let duration: number = 100;
-
-	// Local Settings
-	let elemModal: HTMLElement;
-	let dialogValue: string;
-
-	// Base Classes
-	const cBaseBackdrop: string = 'fixed top-0 left-0 right-0 bottom-0 z-[999] flex justify-center items-center p-4';
-	const cBaseDialog: string = 'p-4 w-full space-y-4 rounded-xl drop-shadow';
-	const cBaseHeader: string = 'flex justify-start items-center space-x-4';
-	const cBaseIcon: string = 'fill-black dark:fill-white bg-primary-500/20 flex justify-center items-center w-10 mx-auto aspect-square rounded-full';
-	const cBaseImage: string = 'w-full h-auto rounded-lg';
-	const cBaseFooter: string = 'flex justify-end space-x-4';
-
-	// Set the Result value response based on button selection
-	function setResult(v: any): void {
-		if ($dialogStore[0].result) {
-			$dialogStore[0].result(v);
-		}
+	/** Provide classes to style the dialog width. */
+	export let width: string = 'w-full max-w-[640px]';
+	/** Provide classes to style the dialog. */
+	export let height: string = 'h-auto';
+	/** Provide classes to style the dialog padding. */
+	export let padding: string = 'p-4';
+	/** Provide classes to style the dialog spacing. */
+	export let spacing: string = 'space-y-4';
+	/** Provide classes to style the dialog border radius. */
+	export let rounded: string = 'rounded-container-token';
+	/** Provide classes to style dialog box shadow. */
+	export let shadow: string = 'shadow-xl';
+
+	// Props (buttons)
+	/** Provide classes for neutral buttons, such as Cancel. */
+	export let buttonNeutral: string = 'btn-ghost-surface';
+	/** Provide classes for positive actions, such as Confirm or Submit. */
+	export let buttonPositive: string = 'btn-filled-primary';
+	/** Override the text for the Cancel button. */
+	export let buttonTextCancel: string = 'Cancel';
+	/** Override the text for the Confirm button. */
+	export let buttonTextConfirm: string = 'Confirm';
+	/** Override the text for the Submit button. */
+	export let buttonTextSubmit: string = 'Submit';
+
+	// Props (regions)
+	/** Provide classes to style the backdrop. */
+	export let regionBackdrop: string = 'bg-backdrop-token';
+	/** Provide arbitrary classes to dialog header region. */
+	export let regionHeader: string = 'text-2xl font-bold';
+	/** Provide arbitrary classes to dialog body region. */
+	export let regionBody: string = 'max-h-[200px] overflow-hidden';
+	/** Provide arbitrary classes to dialog footer region. */
+	export let regionFooter: string = 'flex justify-end space-x-2';
+
+	// Base Styles
+	const cBackdrop: string = 'fixed top-0 left-0 right-0 bottom-0 z-[999] flex justify-center items-center p-4';
+	const cDialogImage: string = 'w-full h-auto';
+
+	// Local
+	let promptValue: any;
+
+	// Dialog Store Subscription
+	dialogStore.subscribe((dArr: any[]) => {
+		if (!dArr.length) return;
+		// Set the local dialog value (for prompt)
+		promptValue = dArr[0].value;
+	});
+
+	// Event Handlers ---
+
+	function onBackdropClick(e: any): void {
```

Comment:
`MouseEvent`

Reply:
Refuses to allow for `e.target.classList` when I do this, even though the value is there when logged. @ryceg I might suggest testing these changes directly before suggesting theme.

The goal here is we need to ensure the click is on the dialog backdrop, and not the modal window. If so, then allow the close to trigger. I tried rewriting to use `stopPropigation()` but no dice. So I'm open to suggestions.

Reply:
Sorry- was shooting from the hip. The issue is essentially that `e.target` is of the `EventTarget` type, which is not limited to `Element`s; other objects can be event targets, too. For example [XMLHttpRequest](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest), [AudioNode](https://developer.mozilla.org/en-US/docs/Web/API/AudioNode), and [AudioContext](https://developer.mozilla.org/en-US/docs/Web/API/AudioContext) are also event targets. *

Given that those types do _not_ have a `classList`, in order to keep dialog from breaking on those (admittedly weird) outliers, we'll need to test to make sure that the target is of an appropriate type. Sling a `if (!(e.target instanceof Element)) return;` in there, and we should be golden. 

This also is the same pattern that is present in `AccordionGroup` :

`if (!(e.currentTarget instanceof HTMLDetailsElement)) return;`


and also in `DocsAppBar.svelte` (line 35):

`if (!(e.target instanceof HTMLInputElement)) return;`


*: [copied from mdn](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget)